File: Source/assets/scripts/njliTests/PhysicsConstraintBuilder.lua

--Test the functions for PhysicsConstraintBuilder
local Test = {}

local function PhysicsConstraintBuilder_calculateSerializeBufferSize_()
--//! [PhysicsConstraintBuilder_calculateSerializeBufferSize_]
-- @TODO: Test: PhysicsConstraintBuilder:calculateSerializeBufferSize()
--//! [PhysicsConstraintBuilder_calculateSerializeBufferSize_]

end

local function PhysicsConstraintBuilder_serialize_dataBuffer_btSerializer()
--//! [PhysicsConstraintBuilder_serialize_dataBuffer_btSerializer]
-- @TODO: Test: PhysicsConstraintBuilder:serialize(dataBuffer, btSerializer)
--//! [PhysicsConstraintBuilder_serialize_dataBuffer_btSerializer]

end

local function PhysicsConstraintBuilder_getObjectType_()
--//! [PhysicsConstraintBuilder_getObjectType_]
-- @TODO: Test: PhysicsConstraintBuilder:getObjectType()
--//! [PhysicsConstraintBuilder_getObjectType_]

end

local function PhysicsConstraintBuilder_getClassName_()
--//! [PhysicsConstraintBuilder_getClassName_]
-- @TODO: Test: PhysicsConstraintBuilder:getClassName()
--//! [PhysicsConstraintBuilder_getClassName_]

end

local function PhysicsConstraintBuilder_getType_()
--//! [PhysicsConstraintBuilder_getType_]
-- @TODO: Test: PhysicsConstraintBuilder:getType()
--//! [PhysicsConstraintBuilder_getType_]

end

local function PhysicsConstraintBuilder_createArray_size()
--//! [PhysicsConstraintBuilder_createArray_size]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.createArray(size)
--//! [PhysicsConstraintBuilder_createArray_size]

end

local function PhysicsConstraintBuilder_destroyArray_array()
--//! [PhysicsConstraintBuilder_destroyArray_array]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.destroyArray(array)
--//! [PhysicsConstraintBuilder_destroyArray_array]

end

local function PhysicsConstraintBuilder_create_()
--//! [PhysicsConstraintBuilder_create_]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.create()
--//! [PhysicsConstraintBuilder_create_]

end

local function PhysicsConstraintBuilder_clone_object()
--//! [PhysicsConstraintBuilder_clone_object]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.clone(object)
--//! [PhysicsConstraintBuilder_clone_object]

end

local function PhysicsConstraintBuilder_destroy_object()
--//! [PhysicsConstraintBuilder_destroy_object]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.destroy(object)
--//! [PhysicsConstraintBuilder_destroy_object]

end

local function PhysicsConstraintBuilder_load_object_L_stack_index()
--//! [PhysicsConstraintBuilder_load_object_L_stack_index]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.load(object, L, stack_index)
--//! [PhysicsConstraintBuilder_load_object_L_stack_index]

end

local function PhysicsConstraintBuilder_type_()
--//! [PhysicsConstraintBuilder_type_]
-- @TODO:Test: NJLI.PhysicsConstraintBuilder.type()
--//! [PhysicsConstraintBuilder_type_]

end


function Test:testAll()
	PhysicsConstraintBuilder_calculateSerializeBufferSize_()
	PhysicsConstraintBuilder_serialize_dataBuffer_btSerializer()
	PhysicsConstraintBuilder_getObjectType_()
	PhysicsConstraintBuilder_getClassName_()
	PhysicsConstraintBuilder_getType_()
	PhysicsConstraintBuilder_createArray_size()
	PhysicsConstraintBuilder_destroyArray_array()
	PhysicsConstraintBuilder_create_()
	PhysicsConstraintBuilder_clone_object()
	PhysicsConstraintBuilder_destroy_object()
	PhysicsConstraintBuilder_load_object_L_stack_index()
	PhysicsConstraintBuilder_type_()
end

return Test

File: Source/jli_game_engine/version-1.0/src/PhysicsConstraintBuilder.cpp
//
//  PhysicsConstraintBuilder.cpp
//  JLIGameEngineTest
//
//  Created by James Folk on 6/15/15.
//  Copyright (c) 2015 James Folk. All rights reserved.
//

#pragma mark Implementation

#include "PhysicsConstraintBuilder.h"
#include "JLIFactoryTypes.h"
#include "World.h"

#define TAG "PhysicsConstraintBuilder.cpp"
#define FORMATSTRING "{\"njli::PhysicsConstraintBuilder\":[]}"
#include "btPrint.h"

namespace njli
{
    PhysicsConstraintBuilder::PhysicsConstraintBuilder()
    {
    }
    
    PhysicsConstraintBuilder::PhysicsConstraintBuilder(const PhysicsConstraintBuilder &copy)
    {
        
    }
    
    PhysicsConstraintBuilder::~PhysicsConstraintBuilder()
    {
    }
    
    PhysicsConstraintBuilder &PhysicsConstraintBuilder::operator=(const PhysicsConstraintBuilder &rhs)
    {
        if (this != &rhs)
        {
            
        }
        return *this;
    }
    
    s32	PhysicsConstraintBuilder::calculateSerializeBufferSize() const
    {
        return 0;
    }
    
    void PhysicsConstraintBuilder::serialize(void* dataBuffer, btSerializer* serializer) const
    {
        
    }
    
    u32 PhysicsConstraintBuilder::getObjectType()const
    {
        return JLI_OBJECT_TYPE_PhysicsConstraint;
    }
    
    const char *PhysicsConstraintBuilder::getClassName()const
    {
        return "PhysicsConstraintBuilder";
    }
    
    s32 PhysicsConstraintBuilder::getType()const
    {
        return PhysicsConstraintBuilder::type();
    }
    
    PhysicsConstraintBuilder::operator std::string() const
    {
        return njli::JsonJLI::parse(string_format("%s", FORMATSTRING).c_str());
    }
    
    PhysicsConstraintBuilder **PhysicsConstraintBuilder::createArray(const u32 size)
    {
        return (PhysicsConstraintBuilder**)World::getInstance()->getWorldFactory()->createArray(PhysicsConstraintBuilder::type(), size);
    }
    
    void PhysicsConstraintBuilder::destroyArray(PhysicsConstraintBuilder **array, const u32 size)
    {
        World::getInstance()->getWorldFactory()->destroyArray((AbstractFactoryObject**)array, size);
    }
    
    PhysicsConstraintBuilder *PhysicsConstraintBuilder::create()
    {
        return dynamic_cast<PhysicsConstraintBuilder*>(World::getInstance()->getWorldFactory()->create(PhysicsConstraintBuilder::type()));
    }
    
    PhysicsConstraintBuilder *PhysicsConstraintBuilder::clone(const PhysicsConstraintBuilder &object)
    {
        return dynamic_cast<PhysicsConstraintBuilder*>(World::getInstance()->getWorldFactory()->clone(object, false));
    }
    
    void PhysicsConstraintBuilder::destroy(PhysicsConstraintBuilder *object)
    {
        if(object)
        {
            World::getInstance()->getWorldFactory()->destroy(object);
        }
    }
    
    void PhysicsConstraintBuilder::load(PhysicsConstraintBuilder &object, lua_State *L, int index)
    {
        // Push another reference to the table on top of the stack (so we know
        // where it is, and this function can work for negative, positive and
        // pseudo indices
        lua_pushvalue(L, index);
        // stack now contains: -1 => table
        lua_pushnil(L);
        // stack now contains: -1 => nil; -2 => table
        while (lua_next(L, -2))
        {
            // stack now contains: -1 => value; -2 => key; -3 => table
            // copy the key so that lua_tostring does not modify the original
            lua_pushvalue(L, -2);
            // stack now contains: -1 => key; -2 => value; -3 => key; -4 => table
            const char *key = lua_tostring(L, -1);
            const char *value = lua_tostring(L, -2);
            if(lua_istable(L, -2))
            {
                PhysicsConstraintBuilder::load(object, L, -2);
            }
            else
            {
                if(lua_isnumber(L, index))
                {
                    double number = lua_tonumber(L, index);
                    printf("%s => %f\n", key, number);
                }
                else if(lua_isstring(L, index))
                {
                    const char *v = lua_tostring(L, index);
                    printf("%s => %s\n", key, v);
                }
                else if(lua_isboolean(L, index))
                {
                    bool v = lua_toboolean(L, index);
                    printf("%s => %d\n", key, v);
                }
                else if(lua_isuserdata(L, index))
                {
                    //                    swig_lua_userdata *usr;
                    //                    swig_type_info *type;
                    //                    assert(lua_isuserdata(L,index));
                    //                    usr=(swig_lua_userdata*)lua_touserdata(L,index);  /* get data */
                    //                    type = usr->type;
                    //                    njli::AbstractFactoryObject *object = static_cast<njli::AbstractFactoryObject*>(usr->ptr);
                    //                    printf("%s => %d:%s\n", key, object->getType(), object->getClassName());
                    
                }
            }
            // pop value + copy of key, leaving original key
            lua_pop(L, 2);
            // stack now contains: -1 => key; -2 => table
        }
        // stack now contains: -1 => table (when lua_next returns 0 it pops the key
        // but does not push anything.)
        // Pop table
        lua_pop(L, 1);
        // Stack is now the same as it was on entry to this function
    }
    
    u32 PhysicsConstraintBuilder::type()
    {
        return JLI_OBJECT_TYPE_PhysicsConstraintBuilder;
    }
}